docs/tutorials/04_stream/viz_interaction.py

```diff
@@ -0,0 +1,175 @@
+
+Notes
+------
+For this example your python version should be 3.8 or greater
+
+
+"""
+
+from fury.stream.server.server import WEBRTC_AVAILABLE
+from fury import actor, window
+import numpy as np
+
+import multiprocessing
+# if this example it's not working for you and you're using MacOs
+# uncoment the following line
+# multiprocessing.set_start_method('spawn')
+from fury.stream.server import web_server
+from fury.stream.client import FuryStreamClient, FuryStreamInteraction
+try:
+    import aiortc
+    WEBRTC_AVAILABLE = True
+except ImportError:
+    WEBRTC_AVAILABLE = False
+
+
+if __name__ == '__main__':
+    interactive = False
+    # `use_raw_array` is a flag to tell the server to use python RawArray
+    # instead of SharedMemory which is a new feature in python 3.8
+    # https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Array
+    # https://docs.python.org/3/library/multiprocessing.html#shared-memory-objects
+    #
+    use_raw_array = False
+   
+    window_size = (300, 300)
+    # `max_window_size` are the maximum size of the window that will be
+    # allowed to be sent to the browser. For example, if you set
+    # `max_window_size=(800, 800)` then the browser will be limited to
+    # a window of size (800, 800).
+    #
+    max_window_size = (400, 400)
+    # 0 ms_stream means that the frame will be sent to the server
+    # right after the rendering
+
+    # `ms_interaction` is the time in milliseconds that the user will have
+    # to interact with the visualization
+    #
+    ms_interaction = 1
+    # `ms_stream` is the number of milliseconds that the server will
+    # wait before sending a new frame to the browser. If `ms_stream=0`
+    # then the server will send the frame right after the rendering.
+    #
+    ms_stream = 0
+    # max number of interactions to be stored inside the queue
+    max_queue_size = 17
+    ######################################################################
+    centers = np.array([
+        [0, 0, 0],
+        [-1, 0, 0],
+        [1, 0, 0]
+    ])
+    colors = np.array([
+        [1, 0, 0],
+        [0, 1, 0],
+        [0, 0, 1]
+    ])
+
+    actors = actor.sphere(
+        centers, opacity=.5, radii=.4, colors=colors)
+    scene = window.Scene()
+
+    scene.add(actors)
+
+    showm = window.ShowManager(
+        scene, size=(window_size[0], window_size[1])
+    )
+
+    showm.initialize()
+
+    stream = FuryStreamClient(
+        showm, max_window_size=max_window_size,
+        use_raw_array=use_raw_array)
+    stream_interaction = FuryStreamInteraction(
+        showm, max_queue_size=max_queue_size,
+        use_raw_array=use_raw_array)
+
+    if use_raw_array:
+        p = multiprocessing.Process(
+            target=web_server,
+            args=(
+                stream.img_manager.image_buffers,
+                None,
+                stream.img_manager.info_buffer,
+                None,
+                stream_interaction.circular_queue.head_tail_buffer,
```

Comment:
The args can be simplified very much by just providing as input ``stream`` and ``stream_interaction``. Then you can grab the other properties from the inside your function.

Reply:
Otherwise, create two classes one for shared memory and one for raw array.